apps/protocol-frontend/src/hooks/useContributionMetadataStore.ts

```diff
@@ -0,0 +1,27 @@
+import { useMutation } from '@tanstack/react-query';
+import { INFURA_PROJECT_ID, INFURA_PROJECT_SECRET } from '../utils/constants';
+import { MintedContributionSchemaV1, IPFS } from '@govrn/protocol-client';
+
+const useContributionMetadataStore = () => {
+  const { mutateAsync, isLoading, isError, isSuccess } = useMutation(
+    ['ContributionMetadataStore'],
+    async (data: Omit<MintedContributionSchemaV1, 'schema'>) => {
+      const ipfs = new IPFS(INFURA_PROJECT_ID, INFURA_PROJECT_SECRET);
+      return await ipfs.storeContributionMetadata({
+        ...data,
+      });
+    },
+    {
+      onSuccess: (data, ipfsURI) => {
+        return ipfsURI;
```

Comment:
We probably don't want to display a toast here do we? I'm thinking probably not since this is called during other actions such as the bulk mint. 

Could be too many Toasts, but nice that we'd have that option if we wanted to use this for something else (or add the toast regardless, but feeling like having a Toast for this and another on Bulk Mint success is too much)

Reply:
Good call I can add the toast as a parameter